creator/ingest_requests/models.py

```diff
@@ -1,16 +1,160 @@
+        blank=True,
+        related_name="ingest_request",
+        on_delete=models.CASCADE,
+        help_text="The associated ingest job for this ingest request",
+    )
+
     created_at = models.DateTimeField(
-        null=False, help_text="Time when the ingest_request was created"
+        auto_now_add=True,
+        null=True,
+        help_text="Time when the ingest request was created",
+    )
+
+    creator = models.ForeignKey(
+        User,
+        null=True,
+        blank=True,
+        related_name="ingest_requests",
+        on_delete=models.SET_NULL,
+        help_text="The user who submitted this ingest request",
     )
+
+    @staticmethod
+    def name_from_version_ids(version_ids):
+        """
+        Create IngestRequest.name by joining the version ids together
+        """
+        return DELIMITER.join([NAME_PREFIX] + version_ids)
+
+    @staticmethod
+    def version_ids_from_name(request_name):
+        """
+        Parse IngestRequest.name into the file version ids which were used
+        to create the name
+        """
+        return request_name.lstrip(NAME_PREFIX).split(DELIMITER)
+
+    @classmethod
+    def find_by_version_ids(cls, version_ids):
+        """
+        Lookup an IngestRequest given a list of Version.kf_id strings
+        """
+        return cls.objects.filter(
+            name=cls.name_from_version_ids(version_ids)
+        ).first()
+
+    def start_job(self, task, *args, **kwargs):
+        """
+        Enqueue an ingest task to start an ingest job
+        """
+        pass
+
+    def cancel_job(self):
+        """
+        Cancel or stop the associated job if it is running
+
+        This method is called when an IngestRequest is deleted
+        """
+        pass
+
+    def clean(self):
+        """
+        Validate model instance and raise a
+        django.core.exceptions.ValidationError if validation fails
+        """
+        if not self._versions_exist():
+            raise ValidationError(
+                {"versions": "An IngestRequest must have at least 1 Version"}
+            )
+
+    @staticmethod
+    def _pre_save_hook(sender, instance, *args, **kwargs):
+        """
+        Set the IngestRequest.name using the version ids
+        """
+        # If no versions have been added yet, do nothing
+        if not instance._versions_exist():
+            return
+
+        # Populate the name from the version ids
+        instance.name = IngestRequest.name_from_version_ids(
+            [v.kf_id for v in instance.versions.all()]
+        )
+
+    def _versions_exist(self):
+        """
+        Check that an IngestRequest has at least 1 Version associated with it.
+        """
+        # If instance hasn't been saved yet, then there is no possibility
+        # for versions to exist
+        if not self.id:
+            return False
+
+        # If no versions have been added, instance.versions won't exist
+        versions_exist = True
+        try:
+            self.versions
```

Comment:
I believe the more semantic way of doing this would be to check `self.versions.exists()` rather than throwing an exception.
Have you tested this? I would expect `self.versions` to return a queryset that resolves to an empty list rather than throw an exception.

Reply:
Hmm I must have been doing something wrong before bc I was getting this exception, a lot. You are right tho, I just did a quick test and I can just get the queryset and check the length. I wonder if its bc I was playing around in the django shell and didn't have autoreload on.